plugins/pulp_rpm/plugins/importers/yum/repomd/metadata.py

```diff
@@ -277,13 +276,11 @@ def generate_dbs(self):
              filelists.PACKAGE_TAG, filelists.process_package_element),
             (other.METADATA_FILE_NAME, other.PACKAGE_TAG, other.process_package_element),
         ):
-            xml_file_handle = self.get_metadata_file_handle(filename)
-            try:
+            with contextlib.closing(self.get_metadata_file_handle(filename)) as xml_file_handle:
```

Comment:
This isn't directly related, but while I was using contextlib, I couldn't help myself.